includes/models/model.llms.lesson.php

```diff
@@ -364,16 +364,73 @@ public function get_video() {
 		return $this->get_embed( 'video' );
 	}
 
+	/**
+	 * Get the lesson prerequisite
+	 *
+	 * @return int [ID of the prerequisite post]
+	 *
+	 * @since 3.37.1 added filter and compatibility for course level prerequisite setting
+	 */
+	public function get_prerequisite() {
+
+		$pre_req = false;
+
+		$course = $this->get_course();
+
+		// Check course level setting
+		if ( $course && $course->must_complete_sequentially() ){
+			// might be false so we are okay
+			$pre_req = $this->get_previous_lesson();
+
+		} else if( $this->has_prerequisite ){
+			// Good ol' fashioned way
+			$pre_req = $this->prerequisite;
+
+		}
+
+		/**
+		 * Allow plugins to modify the prerequisite property.
+		 *
+		 * @since 3.37.1
+		 *
+		 * @param $pre_req int|false the ID of the prerequisite or false if none is available
+		 * @param $this LLMS_Lesson the lesson object
+		 */
+		return apply_filters( 'llms_lesson_get_prerequisite', $pre_req, $this );
+	}
+
 	/**
 	 * Determine if lesson prereq is enabled and a prereq lesson is selected
 	 *
 	 * @return   boolean
 	 * @since    3.0.0
+	 * @cince    3.37.1 Add filter and compatibility for course level prerequisite setting.
 	 * @version  3.0.0
```

Comment:
```suggestion
	 * @since 3.0.0
	 * @since [version] Add filter and compatibility for course level prerequisite setting.
	 *
	 * @return bool
```
and so on...